commons/ava_services/gecko_service_initializer_core.go

```diff
@@ -98,14 +146,85 @@ func (g GeckoServiceInitializerCore) GetStartCommand(publicIpAddr string, depend
 		for _, service := range avaDependencies {
 			socket := service.GetStakingSocket()
 			socketStrs = append(socketStrs, fmt.Sprintf("%s:%d", socket.GetIpAddr(), socket.GetPort().Int()))
+			if g.stakingTlsEnabled {
+				// We hardcode the first bootstrapper ID from the TLS identities in gecko_service_tls_identities
+				commandList = append(commandList, "--bootstrap-ids=" + STAKER_1_NODE_ID)
+				// You only need one bootstrapper dependency, and we only currently have one cert -> ID mapping so break for loop here.
+				break
+			}
 		}
 		joinedSockets := strings.Join(socketStrs, ",")
 		commandList = append(commandList, "--bootstrap-ips=" + joinedSockets)
 	}
 	logrus.Debugf("Command list: %+v", commandList)
-	return commandList
+	return commandList, nil
 }
 
 func (g GeckoServiceInitializerCore) GetServiceFromIp(ipAddr string) services.Service {
 	return GeckoService{ipAddr: ipAddr}
 }
+
+
+func getServiceCertAndKeyFiles(cert *x509.Certificate, ca *x509.Certificate) (certFile *bytes.Buffer, keyFile *bytes.Buffer, err error) {
+	certPrivKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		return nil, nil, stacktrace.Propagate(err, "")
+	}
+	certBytes, err := x509.CreateCertificate(rand.Reader, cert, ca, &certPrivKey.PublicKey, certPrivKey)
```

Comment:
nit: can we add parentheses to `&certPrivKey.PublicKey` to make it clearer that we're taking the pointer of the `PublicKey` (I think would be `&(certPrivKey.PublicKey)`)? I was a bit confused looking at this, wondering why we were doing `(&certPrivKey).PublicKey` (which is the same thing as certPrivKey.PublicKey)

Reply:
done